builtin/builtin.ts

```diff
@@ -730,7 +730,14 @@ export interface BaseTypeInfo {
 export interface StructTypeInfo extends BaseTypeInfo {
   kind: TypeKind.Struct;
   methods: Set<string>;
-  constructor: new (...args: any[]) => any;
+  constructor?: new (...args: any[]) => any;
+  fields?: Set<string>; // Field names for struct types
+}
+
+// Interface for ad-hoc struct type assertions that don't require a constructor
+export interface AdHocStructTypeInfo extends BaseTypeInfo {
```

Comment:
Since constructor? is already optional we should use StructTypeInfo and delete AdHocStructTypeInfo